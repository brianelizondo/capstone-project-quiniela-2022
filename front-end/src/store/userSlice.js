import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
    name: 'user',
    initialState: {
        id: null, 
        firstName: null, 
        lastName: null, 
        email: null, 
        username: null, 
        isAdmin: false,
        token: null,
        isAuthenticated: false
    },
    reducers: {
        login: (state, action) => {
            state.id = action.payload.user.id;
            state.firstName = action.payload.user.firstName;
            state.lastName = action.payload.user.lastName;
            state.email = action.payload.user.email;
            state.username = action.payload.user.username;
            state.isAdmin = action.payload.user.isAdmin;
            state.token = action.payload.token;
            state.isAuthenticated = true;
        },
        update: (state, action) => {
            state = action.payload
        },
        logout: (state) => {
            state.id = null;
            state.firstName = null;
            state.lastName = null;
            state.email = null;
            state.username = null;
            state.isAdmin = null;
            state.token = null;
            state.isAuthenticated = false;
        }
    }
})
  
// Action creators are generated for each case reducer function
export const { login, update, logout } = userSlice.actions;
export default userSlice.reducer;