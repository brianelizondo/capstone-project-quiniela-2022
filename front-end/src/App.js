import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import './App.css';

// import APIFootball api
import APIFootball from "./api-football";

// redux dispatch and reducer
import { useDispatch } from 'react-redux';
import { logout } from './store/userSlice';

// Import components
import NavBar from './Navbar';
import Home from './Home';
import QuinielaDetails from './QuinielaDetails';
import GroupList from './GroupList';
import GroupDetails from './GroupDetails';
import MatchList from './MatchList';
import MatchDetails from './MatchDetails';
import TeamList from './TeamList';
import TeamDetails from './TeamDetails';
import Rules from './Rules';

import RegisterForm from './RegisterForm';
import LoginForm from './LoginForm';

import UserProfile from './UserProfile';
import UserMatchUpdate from './UserMatchUpdate';
import UserQuinielaAdd from './UserQuinielaAdd';

import ProtectedRoute from './ProtectedRoute';
import PageNotFound from './PageNotFound';
import Footer from './Footer';

function App(){
    // redux selector and dispatch
    const dispatch = useDispatch();

    // functions to register, update, login, logout users
    const userRegister = async user => {
        try{
            let resp = await APIFootball.userRegister(user);
            return resp;
        }catch (err){
            return err;
        }
    }
    // function to authenticate the user
    const userAuthenticate = async user => {
        try{
            let resp = await APIFootball.userAuthenticate(user);
            return resp;
        }catch (err){
            return err;
        }
    }
    // function to logout the user
    const userLogout = () => {
        dispatch(logout());
    }
    // check if username/email already exists
    const checkUsernameEmail = async (email, username) => {
        try{
            let resp = await APIFootball.checkUsernameEmail(username, email);
            return resp;
        }catch (err){
            return err;
        }
    }

    
    return (
        <BrowserRouter>
            <NavBar userLogout={userLogout} />
            <Switch>
                <Route exact path="/">
                    <Home />
                </Route>
                <Route exact path="/quinielas">
                    <Home />
                </Route>
                
                <Route exact path="/groups">
                    <GroupList />
                </Route>
                <Route exact path="/groups/:group">
                    <GroupDetails />
                </Route>
                <Route exact path="/matches">
                    <MatchList />
                </Route>
                <Route exact path="/matches/phase/:phase/match/:matchID">
                    <MatchDetails />
                </Route>
                <Route exact path="/teams">
                    <TeamList />
                </Route>
                <Route exact path="/teams/:shortname">
                    <TeamDetails />
                </Route>
                <Route exact path="/rules">
                    <Rules />
                </Route>

                <Route exact path="/register">
                    <RegisterForm userRegister={userRegister} checkUsernameEmail={checkUsernameEmail} />
                </Route>
                <Route exact path="/login">
                    <LoginForm userAuthenticate={userAuthenticate} />
                </Route>

                {/* ROUTES TO PROTECT LATER */}
                <ProtectedRoute exact path="/quinielas/:username/:quinielaID" component={QuinielaDetails} />
                <ProtectedRoute exact path="/users/:username/profile" component={UserProfile} />
                <ProtectedRoute exact path="/users/matches/update" component={UserMatchUpdate} />
                <ProtectedRoute exact path="/users/:username/quinielas/add" component={UserQuinielaAdd} />
                <ProtectedRoute exact path="/users/:username/quinielas/:quinielaID/edit" component={UserQuinielaAdd} />

                <Route exact path="/404">
                    <PageNotFound />
                </Route>
                <Route>
                    <PageNotFound />
                </Route>
            </Switch>
            <Footer />
        </BrowserRouter>
    );
}

export default App;
