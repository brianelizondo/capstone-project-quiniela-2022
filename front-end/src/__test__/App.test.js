import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { persistStore } from 'redux-persist';
import store from '../store/store';
import App from '../App';
let persistor = persistStore(store);

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'App' component without crashing", async () => {
        render(
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <App />
                </PersistGate>
            </Provider>
        );
        const textFound = screen.getByText("Quinielas World Cup 2022");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'App' matches snapshot", () => {
        const {asFragment} = render(
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistor}>
                    <App />
                </PersistGate>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});