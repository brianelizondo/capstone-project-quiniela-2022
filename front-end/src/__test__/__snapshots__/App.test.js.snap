// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'App' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <div
      class="justify-content-center nav"
    >
      <div
        class="nav-item"
      >
        <a
          class="nav-link active"
          data-rr-ui-event-key="/"
          href="/"
        >
          Quinielas
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/groups"
          href="/groups"
        >
          Groups
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/matches"
          href="/matches"
        >
          Matches
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/teams"
          href="/teams"
        >
          Teams
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rr-ui-event-key="/rules"
          href="/rules"
        >
          Rules
        </a>
      </div>
      <div
        class="nav-item dropdown"
      >
        <a
          aria-expanded="false"
          class="dropdown-toggle nav-link"
          href="#"
          id="nav-dropdown"
          role="button"
          tabindex="0"
        >
          Login/Register
        </a>
      </div>
    </div>
  </div>
  <div
    class="Home col-md-8 offset-md-2"
  >
    <h1>
      Quinielas World Cup 2022
    </h1>
    <p>
      List of active and participating quinielas
    </p>
    <div
      class="Home-quiniela-list"
    />
  </div>
  <div
    class="Footer"
  >
    Quinielas World Cup 2022. Developed by 
    <a
      href="https://github.com/brianelizondo"
    >
      Brian Elizondo
    </a>
     for Capstone Project
  </div>
</DocumentFragment>
`;
